### DISCUSSION TITLE
Version 1.7+ Violates BSD License

### DISCUSSION THREAD

**Discussion by fazekasj** [fnc12/sqlite_orm#905] [General]
The license for this code was changed either without maintaining the existing requirements under the BSD 3 license or removing all of the code licensed under BSD 3.  As I understand it, this essentially makes all changes after 1.6 in violation of the original license and I'd guess most users won't be able to use it legally.

Does anyone know if there is a fork for this project that is maintain licenses correctly?

**Comment by fnc12** [comment]
> The license for this code was changed

The license is a part of the repo. It means that it can be changed by any commit just like any file located inside repo: README, tests code, CMakeLists etc. BSD 3 license still exists in all older versions till (including) v1.6 so please feel free to use it without paying me a huge amount of 50$. 

If you still believe that the license was violated, corrupted, injured, raped or whatever you imagine can be done by a bald headed software developer from Almaty like me please go to [releases page](https://github.com/fnc12/sqlite_orm/releases), download v1.6 release or any earlier (really any of them, I've checked them out before writing in case of hacker attack you know), open LICENSE file and ensure that it is BSD3 with no changes. What does it mean? It means that you (and not only you actually) can use these versions within any project, even within a paid project, even if your project makes 1M$ daily income you still can use it for free. And I shall not go to your house to ask for some money (I promise).

v1.7 has different license. What does it mean? It means that all code written after license change (please notice the word 'after', it is important) is covered with a different license. This is the same story when you use any vendor's service like Apple Music, Netflix or whatever and one day you see that you need to accept new terms of use because terms of use were changed by vendor. Again: it doesn't mean that you must accept new terms. You must do it only if you want to use it more. `sqlite_orm` license change is something similar but you can still use old versions with the old license (yay, even better than Netflix!).

> I'd guess most users won't be able to use it legally

There are a lot of users who is satisfied with v1.6 cause it has the most of features. But there are also a lot of users who paid for the newest version(s) a huge (HUGEST!) amount of money of 50$. People understand what they pay for: they motivate me to maintain this lib without losing interest. Also there are a lot of people who says 'thanks for such a great library' every day! I really appreciate these words!

> Does anyone know if there is a fork for this project that is maintain licenses correctly?

Sounds like a provocative question cause when someone wants to inspect forks he or she goes to 'Insights' tab, 'Network' section and looks through existing forks. This is the correct way of inspecting forks. But when someone (I mean you @fazekasj ) asks about forks looks like he or she wants someone (me!) to notice him or her. Ok I noticed you. Probably you are a person who likes the attention. It is ok. People like you exist in this world and I have nothing against them.
If you want to make a fork you can do it by yourself (I mean it!). Just go to the right top corner, click 'fork' and you're done! GitHub magic. Clicking this button is free BTW. But don't forget the license of course. Pay attention to 'LICENSE' file of forked commit.

Eventually you look like a person who likes to go by bus without payment. Again: people like you exist and it is ok. No one tells you that you must pay 50$ right now. Payments for open source development is a very important thing. It makes developers more motivated. If you don't believe it probably you never tried to do it.

**Comment by fazekasj** [comment]
Disclaimer: IANAL

You don't seem to understand my question/concern.  You can't simply remove a license from previously licensed code.  If you could then anyone could copy all the code and remove the license.  

This code, including version 1.7 is licensed under the previous BSD license, you cannot simply remove it because you changed your mind.  1.7 now also has subsequent licenses, that also conflict with each other.  Code can't be licensed under AGPL and MIT licenses at the same time.  

Even if someone paid you 50$, you can't just remove the AGPL license.  This means that even if I or anyone made this contribution, all of the GPL requirements would still need to be maintained in order to remain in GPL compliance.  Therefore, the user would be subject to maintaining three separate licenses that have conflicting clauses, which is legally impossible.

At this point, it appears that from Sep 28, 2021 forward code from this repository can't be legally included in any code.

**Comment by fnc12** [comment]
> You don't seem to understand my question/concern.

I did. You did not understand that I understand it.

> You can't simply remove a license from previously licensed code.

No one removed the license from previous code. Previous license is still available inside previous commits. This is how git works.

> This code, including version 1.7 is licensed under the previous BSD license

No. v1.7 is not licensed under BSD. It is documented inside git commits.

> you cannot simply remove it because you changed your mind

Man I can do whatever I want with any content created by me in the internet. This is how it works.

> Code can't be licensed under AGPL and MIT licenses at the same time.

Lol what? It can. I see that you are not a lawyer. Google 'multi-licensing' and check how popular open source projects make money. This is what I used to study before applying license changes.

> Even if someone paid you 50$, you can't just remove the AGPL license.

No one removes the license. What are you talking about? Are you a troll?

> Therefore, the user would be subject to maintaining three separate licenses that have conflicting clauses, which is legally impossible.

Why are you trying to argue about license not being a lawyer? Just lurk more info if you don't know something.

P.S. I see that you registered your account today and the only thing you did here on GitHub is posting this discussion.

**Reply by fazekasj** [reply]
That's an interesting legal interpretation.  Simply stating that 1.7 isn't licensed under BSD doesn't make it true.  The license needs to be carried forward from the previous work, as required by the license, for all code in this repo.  As the owner, well sort of, I see many contributors you would need to get permission from to change the license, you aren't going to start legal action against yourself so in practice you can release the code under different licenses.  But someone looking at this project for the first time in 1.7 might not be aware of those previous requirements that apply to all version of code in the repo.

I can see that your legal advice is the result of some google searches, I recommend you seek some legitimate legal council if you want to see what you can do to save/monetize this project.

**Reply by fnc12** [reply]
Why do you give me advices while I did no ask you to do it? Look at the situation: there are two humans with different opinions. Both are not lawyers but one of them (me) has read about 10k lines of information about open source software licensing and other (you) did no do that. It means that I have more experience in that topic than you. But it is you who gives advices not me. Looks like a perfect case of Dunning–Kruger effect.
Anyway what is your goal?

**Reply by fazekasj** [reply]
I'd like to use this library and also see it continue to be used and maintained.  Unfortunately, in it's current state I can't see this being acceptable to anyone who doesn't want to comply with restrictive GPL requirements.

**Reply by fnc12** [reply]
Ok. If you don't like AGPL license then use v1.6. It has nothing common with AGPL. It is the latest release under BSD3 license

**Comment by fnc12** [comment]
@fazekasj FYI if you want to know what I know go and read all the articles from this list https://github.com/mrjoelkemp/awesome-paid-open-source
